DSESwift/DataStorageEngine.swift

```diff
@@ -31,12 +30,12 @@ public class DataStorageEngine{
     // reference to the datasyncer
     var dataSyncer = DataSyncer()
```

Comment:
Ah I misunderstood. You meant the variable "dataSyncer". Yes it should be private within the scope of production code, but having it as internal variable makes it easier to check the state of data syncer in the tests. We like to keep our variables and utility functions private, but we’re not fanatic about it. Sometimes we need to make a variable or utility function protected so that it can be accessed by a test. For us, tests rule. If a test in the same package needs to call a function or access a variable, we’ll make it protected or package scope. However, we’ll first look for a way to maintain privacy. Loosening encapsulation is always a last resort.